Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -12,16 +12,17 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIOConcurrencyHelpers
 import NIOCore
 
 // MARK: NIOPCAPRingBuffer
 /// Storage for the most recent set of packets captured subject to constraints.
 /// Use ``addFragment(_:)`` as the sink to a ``NIOWritePCAPHandler`` and call ``emitPCAP()``
 /// when you wish to get the recorded data.
 /// - Warning:  This class is not thread safe so should only be called from one thread.
-public class NIOPCAPRingBuffer {
-    private var pcapFragments: CircularBuffer<ByteBuffer>
-    private var pcapCurrentBytes: Int
+public final class NIOPCAPRingBuffer: Sendable {
```

Comment:
This is a good question. This class is only used in tests by us (although it's public API).

I was trying to get rid of a warning where a `NIOPCAPRingBuffer` is captured in an `ELF/map` chained after a call to `write` on `ChannelHandlerContext` (the test in question is in `TriggerOnCumulativeSizeHandler/TriggerOnCumulativeSizeHandler/write`):

```swift
let boundedSink = NIOLoopBound(self.sink, eventLoop: context.eventLoop)
context.write(data).map { [pcapRingBuffer] in
  boundedSink.value(captureBytes(ringBuffer: pcapRingBuffer))
}.cascade(to: promise)
```

This piece of code also seems like a somewhat sensible pattern to do as a client when you're trying to debug something (which is why this class exists as public API, I assume). I'm not sure how to refactor this to get rid of the warning without making the class `Sendable`, as `pcapRingBuffer ` is modified as a result of the call to `context.write()`. This means I can't call `captureBytes(ringBuffer:)` outside of the closure, as we'd trigger the PCAP at the wrong time (causing the wrong behaviour and assertion failures).

I'm open to refactor suggestions if we want to avoid making this `Sendable`. Having said that, is there any reason why you would suggest avoiding it, Cory? Or would we be better off making these changes?